src/components/TokenClaimDialog.tsx

```diff
@@ -319,6 +355,43 @@ function useRewardClaims() {
     },
     [chainId, account, rewardsContract, updateClaimStatus],
   )
+
+  const claimGaugeReward = useCallback(
+    async (gauge: {
+      poolName: string | undefined
+      address: string
+      rewards: GaugeReward[]
+    }) => {
+      if (!chainId || !account || !minterContract || !library) return
+      try {
+        updateClaimStatus(gauge?.poolName || "", STATUSES.PENDING)
+        if (gauge.rewards.length > 0 && gauge.address) {
+          const liquidityGaugeContract = getContract(
+            gauge.address,
+            LIQUIDITY_GAUGE_V5_ABI,
+            library,
+            account,
+          ) as LiquidityGaugeV5
+          const txn = await liquidityGaugeContract["claim_rewards(address)"](
```

Comment:
it is possible to fire `claim_rewards` and `mint` in the same block so the user can approve 2 txns back to back instead of waiting for one to be mined before signing the other?